### DISCUSSION TITLE
'pyodide' This package does not appear to be installable

### DISCUSSION THREAD

**Discussion by ChinaGodzilla** [pyscript/pyscript#1184] [General]
I just started to learn the source code in the 'pyscript' tutorial for newcomers, I don't quite understand, 'from pyodide.http import open_url' pycharm red prompts that there is no such package, 'pip install pyodide', it cannot be installed.

![image](https://user-images.githubusercontent.com/43966525/218302412-9e9eb783-ac71-457f-ba2c-107aad34c344.png)

![image](https://user-images.githubusercontent.com/43966525/218302600-e52ca24b-2cfa-4d80-92ec-ebbee2513e51.png)

Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting pyodide
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/87/f6/a828d76fbb3fab2c114f2c5f6cd045b574e7c52d2f192c112cc97c23c0f4/pyodide-0.0.2.tar.gz (19 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\Administrator\AppData\Local\Temp\pip-install-8ce5bkpv\pyodide_87e82d8dc1ee4ae2a61e56190ec35279\setup.py", line 7, in <module>
          raise ValueError(
      ValueError: Pyodide is a Python distribution that runs in the browser or Node.js. It cannot be installed from PyPi.
                  See https://github.com/pyodide/pyodide for how to use Pyodide.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by FabioRosado** [comment]
Hello @ChinaGodzilla as the exception says, pyodide runs in the browser so it cannot be installed with pypi.   Pyodide is the interpreter that PyScript uses (at least for now, we have plans to add more interpreters like micropython). 

Pyodide is being served from `https://cdn.jsdelivr.net/pyodide/v0.22.1/full/pyodide.js` and due to some **_magic_** you can use pyodide functions in your python code.  I know that you are learning how to use pyscript  and I'm assuming that you want to download pyodide so you can use the IDE suggestions.

In this case, perhaps the best option will be referring to the [Pyodide's API reference documentation](https://pyodide.org/en/stable/usage/api-reference.html).  

Please let us know if  this helps and feel free to ask for any help 😄

**Reply by ChinaGodzilla** [reply]
@FabioRosado thanks you

**Reply by ChinaGodzilla** [reply]
@FabioRosado  

![image](https://user-images.githubusercontent.com/43966525/218309443-9cdcd247-1b2e-4030-853c-d8af7ab5389c.png)


as the picture shows. have code

flavor_elements = Documentation. getElementsByName("flavor")

The front end prompts that js.document is required, how to express it in code?

![image](https://user-images.githubusercontent.com/43966525/218309310-20229285-46b9-443b-9e83-d1a276d8b280.png)

**Comment by FabioRosado** [comment]
You will have to replace `document.getElementsByName` with `js.document.getElementsByName` you can also import document from `js` such as:

```python
from js import document
```

And then it should work normally. The `js` module is part of that pyodide magic that I mentioned before, it creates a proxy for python < - > JS translations 😄

**Comment by ChinaGodzilla** [comment]
@FabioRosado thanks you !
